nowcasting_dataloader/utils/position_encoding.py

```diff
@@ -308,18 +308,19 @@ def normalize_geospatial_coordinates(
     geospatial_coordinates[1] = geospatial_coordinates[1] * 2 - 1
     # Now create a grid of the coordinates
     # Have to do it for each individual example in the batch, and zip together x and y for it
-    to_concat = []
+    solid_tensor = None
     for idx in range(len(geospatial_coordinates[0])):
         x = torch.from_numpy(geospatial_coordinates[0][idx])
         y = torch.from_numpy(geospatial_coordinates[1][idx])
         grid = torch.meshgrid(x, y)
         pos = torch.stack(grid, dim=-1)
         encoded_position = fourier_encode(pos, **kwargs)
         encoded_position = einops.rearrange(encoded_position, "... n d -> ... (n d)")
-        to_concat.append(encoded_position)
+        if solid_tensor is None:
+            solid_tensor = torch.zeros(len(geospatial_coordinates[0]), *encoded_position.size())
```

Comment:
does `torch.zeros()` use the same data type as `encoded_position`?  (to avoid having to cast between data types in line 321).

Reply:
Good catch! Added making it in the same dtype as `x`